test/pkcs12_format_test.c

```diff
@@ -708,6 +721,84 @@ static int test_multiple_contents(void)
     return end_pkcs12_builder(pb);
 }
 
+static int test_jdk_trusted_attr(void)
+{
+    PKCS12_BUILDER *pb = new_pkcs12_builder("jdk_trusted.p12");
+
+    /* Generate/encode */
+    start_pkcs12(pb);
+
+        start_contentinfo(pb);
+
+            add_certbag(pb, CERT1, sizeof(CERT1), ATTRS3);
+
+        end_contentinfo(pb);
+
+    end_pkcs12_with_mac(pb, &mac_default);
+
+    /* Read/decode */
+    start_check_pkcs12_with_mac(pb, &mac_default);
+
+        start_check_contentinfo(pb);
+
+            check_certbag(pb, CERT1, sizeof(CERT1), ATTRS3);
+
+        end_check_contentinfo(pb);
+
+    end_check_pkcs12(pb);
+
+    return end_pkcs12_builder(pb);
+}
+
+static int test_set0_attrs(void)
```

Comment:
We need a test somewhere for `PKCS12_create_ex2()` as well

Reply:
I was using the fact that the pkcs12 app is using PKCS12_create_ex2() as a way to test that? I've added a test to the recipe for that

Reply:
@mattcaswell I've add some PKCS12_create_ex2 tests but for one of them (the pkcs12_create_ex2_failure_test) I want to confirm the bag has been removed...is there an API to retrieve the STACK_OF bags for a given cert?